### ISSUE TITLE
Trigger mention suggestions without leading space

### ISSUE THREAD

**Issue by mattmazzola** [draft-js-plugins/draft-js-plugins#973]
## Use-case/Problem
The existing regex for initiating the suggestion dropdown requires that the mentionTrigger is the first character or is preceded by a space.  

See: `(\\s|^)${escapeRegExp(trigger)}${regExp}`
https://github.com/draft-js-plugins/draft-js-plugins/blob/master/draft-js-mention-plugin/src/mentionSuggestionsStrategy.js#L8

In my use case I wanted to the user to be able to wrap the mention in characters that provide other meaning for our application such as:
`Hi @Matt [@John]`
The mention for @Matt will work because it is preceded by a space, but the @John will not work because it is preceded by a `[`

I thought I could edit the `mentionRegExp` to overcome this, but found out that it only affects the portion after the trigger.

I feel like this variable should allow over riding the whole regex, but for backwards compatibility we could add new option such as `mentionTriggerRegExp` or something that would replace the entire thing and could default to the current implementation. (Although this name is less generic even though it has more responsibility)

Some questions:
**1. Is there anything in the plugin that requires the leading space?**
It seems odd to not allow this specific part of the regex to be over written. Perhaps this is by design and there was a requirement for the space in order to reliably get search text or selection of entity since it was known to be bounded by spaces?  Is there any reason this plugin couldn't support this scenario?

**2.  Is there anyway to configure the plugin to support this?**
Based on information I found, the answer is _no_.  I would either have to fork the plugin or write my own custom implementation to get around the hard coded regex in mentionSuggestionStrategy.  Although maybe I am wrong and some expert has already developed work-around.

**3. If the answer to the above 2 is true, what would you suggest the best approach is?**
I feel like I am starting to hit the limits of what the plugin is designed for and perhaps custom implementation would be better in the long run, but there is a whole bunch of cursor / content selection logic and work arounds in here that seem like the average person would not know unless they hit the issue and I thought re-using this work would be faster. 

For example I also would like to be able to customize behavior of adding space after entity is added. (Disabling it) and don't think this is possible.

TL;DR;
I would like to be able to have the mentionTrigger not required to be preceded by a space to allow my editor to support optional entities wrapped in brackets like: `[@John]`

**Comment by saketsaurabh** [comment]
@mattmazzola were you able to solve this?

**Comment by mattmazzola** [comment]
Hah, yea I switched libraries and built my own. Ultimately came to conclusion that these plugins are made to achieve a specific task and I would end up doing more fighting to force it to behave a way it wasn't designed to.  You can see evidence of this in the naming / configuration objects etc...

**Comment by YoonjiJang** [comment]
+1

I think this will be especially helpful for Japanese or Chinese keyboards since Japanese/Chinese writings don't use word spacing. 🤔

**Comment by m1n0s** [comment]
Faced this as well. Any other solutions except doing a fork of the plugin?

**Comment by waveiron** [comment]
Commenting to re-open this issue - neither https://github.com/draft-js-plugins/draft-js-plugins/pull/1301 nor https://github.com/draft-js-plugins/draft-js-plugins/pull/2165 were merged :(

**Comment by riddhi-24** [comment]
Any update on this?

**Comment by lovro14** [comment]
It this going to be enabled within plugin or we need to fork it and make changes so we can trigger mention suggestions dropdown without leading space ? Thank you !

**Comment by fxOne** [comment]
@lovro14 do you want to create a pull request?
